spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/sessionmanagement/config/OAuth2SessionManagementSecurityConfig.java

```diff
@@ -19,19 +20,18 @@ public class OAuth2SessionManagementSecurityConfig {
 
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.authorizeRequests(authorizeRequests -> authorizeRequests.mvcMatchers("/home")
-            .permitAll()
-            .anyRequest()
-            .authenticated())
-            .oauth2Login(oauthLogin -> oauthLogin.permitAll())
+        http.authorizeHttpRequests(authorizeRequests -> authorizeRequests
+                        .requestMatchers("/home").permitAll()
+                        .anyRequest().authenticated())
+            .oauth2Login(AbstractAuthenticationFilterConfigurer::permitAll)
             .logout(logout -> logout.logoutSuccessHandler(oidcLogoutSuccessHandler()));
         return http.build();
     }
 
     private LogoutSuccessHandler oidcLogoutSuccessHandler() {
         OidcClientInitiatedLogoutSuccessHandler oidcLogoutSuccessHandler = new OidcClientInitiatedLogoutSuccessHandler(this.clientRegistrationRepository);
 
-        oidcLogoutSuccessHandler.setPostLogoutRedirectUri(URI.create("http://localhost:8081/home"));
+        oidcLogoutSuccessHandler.setPostLogoutRedirectUri(URI.create("http://localhost:8081/home").toString());
```

Comment:
I guess can be shortened to oidcLogoutSuccessHandler.setPostLogoutRedirectUri("http://localhost:8081/home");

Reply:
Fixed